mypy/checkstrformat.py

```diff
@@ -518,7 +546,21 @@ def apply_field_accessors(self, spec: ConversionSpecifier, repl: Expression,
 
         # This is a bit of a dirty trick, but it looks like this is the simplest way.
         temp_errors = self.msg.clean_copy().errors
-        dummy = DUMMY_FIELD_NAME + spec.field[len(spec.key):]
+
+        try:
+            first_obj, split_fields = string._string.formatter_field_name_split(
+                spec.field[len(spec.key):])
+            split_fields_str = [
+                ("." + str(field)) if is_attr else
```

Comment:
This list comprehension is a bit difficult to understand. Can you change it into a for loop, to improve readability?

Reply:
Done